src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs

```diff
@@ -0,0 +1,291 @@
+﻿// ----------------------------------------------------------------------------------
+// 
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.Properties;
+using Microsoft.Azure.Management.RecoveryServices.SiteRecovery.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Creates Azure Site Recovery VM NIC IP configuration for A2A replication.
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RecoveryServicesAsrVMNicIPConfig", DefaultParameterSetName = ASRParameterSets.AzureToAzure, SupportsShouldProcess = true)]
+    [Alias("New-ASRVMNicIPConfig")]
+    [OutputType(typeof(IPConfigInputDetails))]
+    public class NewAzureRmAsrVmNicIPConfig : SiteRecoveryCmdletBase
+    {
+        #region Parameters
+        /// <summary>
+        ///    Gets or sets the NIC Id.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = true,
+            HelpMessage = "Specify the ASR NIC GUID.")]
+        public string NicId { get; set; }
+
+        /// <summary>
+        ///    Specify the ASR Replication Protected Item.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = true,
+            HelpMessage = "Specify the ASR Replication Protected Item.")]
+        public ASRReplicationProtectedItem ReplicationProtectedItem { get; set; }
+
+
+        /// <summary>
+        ///    Gets or sets the IP config name.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = true,
+            HelpMessage = "Specify the IP config name.")]
+        public string IpConfigName { get; set; }
+
+        /// <summary>
+        ///    Gets or sets the if the IP config is primary on NIC.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
```

Comment:
switch parameter. This would have been flagged in the PS design review. Could we start the PS design review for these changes as well?

Reply:
given we dont want to support changing ipconfig from primary to secondary and vice versa, is switch paramater the way to go !?

Reply:
bools are not allowed as params through value. Instead should be through switch param. As for the need, that needs to be determined by us. But bool params aren't allowed like this

Reply:
removed this from input params.. used defaulting logic to set this info.